apps/editing-toolkit/editing-toolkit-plugin/block-patterns/class-block-patterns-from-api.php

```diff
@@ -53,16 +38,30 @@ class Block_Patterns_From_API {
 	 */
 	private $core_to_wpcom_categories_dictionary;
 
+	/**
+	 * This is the current editor type. One of `block_editor` (default), `site_editor`.
+	 *
+	 * @var string
+	 */
+	private $editor_type;
+
 	/**
 	 * Block_Patterns constructor.
 	 *
-	 * @param array                $patterns_sources A array of strings, each of which matches a valid source for retrieving patterns.
-	 * @param Block_Patterns_Utils $utils            A class dependency containing utils methods.
+	 * @param string                    $editor_type The current editor. One of `block_editor` (default), `site_editor`.
+	 * @param Block_Patterns_Utils|null $utils       A class dependency containing utils methods.
 	 */
-	public function __construct( $patterns_sources, Block_Patterns_Utils $utils = null ) {
-		$patterns_sources       = empty( $patterns_sources ) ? array( 'block_patterns' ) : $patterns_sources;
-		$this->patterns_sources = empty( array_diff( $patterns_sources, $this->valid_patterns_sources ) ) ? $patterns_sources : array( 'block_patterns' );
-		$this->utils            = empty( $utils ) ? new \A8C\FSE\Block_Patterns_Utils() : $utils;
+	public function __construct( string $editor_type = 'block_editor', Block_Patterns_Utils $utils = null ) {
```

Comment:
Just wondering about potential mid-deploy fatals with this change to the constructor. E.g. if during deployment this file is changed before `full-site-editing-plugin.php`, then this function _might_ receive an array instead of a string in its first param? Would it be worth dropping the `string` type hint and adding an `is_array` check in the constructor to handle the possibility of an array, and then in a follow-up remove the `is_array` check and add back in the type hint (along with deploying the change to `full-site-editing-plugin.php`?

Reply:
Good spotting. Best to be sure here. I hadn't considered that. 

What do you think about a check like:

```php
		if ( ! is_string( $editor_type ) ) {
			$editor_type = 'block_editor';
		}

```